core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/GetTraitsHandler.java

```diff
@@ -34,12 +34,17 @@
 import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
 
 public class GetTraitsHandler implements OutputOperationHandler<GetTraits, Set<StoreTrait>> {
+    private final Set<StoreTrait> storeTraits;
     private Set<StoreTrait> currentTraits;
 
+    public GetTraitsHandler(final Set<StoreTrait> storeTraits) {
+        this.storeTraits = storeTraits;
+    }
+
     @Override
     public Set<StoreTrait> doOperation(final GetTraits operation, final Context context, final Store store) throws OperationException {
         if (!operation.isCurrentTraits()) {
-            return store.getTraits();
+            return storeTraits;
```

Comment:
You haven't protected the Set.
+ testing

Reply:
I believe this has now been fixed

Reply:
👍